bin/copy_to_distributed_table

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+IFS=$'\n\t'
+
+# default values for certain options
+format='text'
+schema='public'
+
+# we'll append to this string to build options list
+options='OIDS false, FREEZE false'
+
+# outputs usage message on specified device before exiting with provided status
+usage() {
+	cat << 'E_O_USAGE' >&"$2"
+usage: copy_to_distributed_table [-BCTHh] [-c encoding] [-d delimiter]
+	[-e escape] [-n null] [-q quote] [-s schema] filename tablename
+
+  B : use binary format for input
+  C : use CSV format for input
+  T : use text format for input
+  H : specifies file contains header line to be ignored
+  h : print this help message
+
+  c : specifies file is encoded using `encoding`
+      Default: the current client encoding
+  d : specifies the character used to separate columns
+      Default: a tab character in text format, a comma in CSV format
+  e : specifies the character used to escape quotes
+      Default: the same as the `quote` value (quotes within data are doubled)
+  n : specifies the string that represents a null value
+      Default: \N in text format, an unquoted empty string in CSV format
+  q : specifies the quoting character to be used when a data value is quoted
+      Default: double-quote
+  s : specifies the schema in which the target table resides
+      Default: 'public'
+E_O_USAGE
+
+	exit $1;
+}
+
+# process flags
+while getopts ':BCc:d:e:Hhn:q:T' o; do
+	case "${o}" in
+		B)
+			format='binary'
+			;;
+		C)
+			format='csv'
+			;;
+		c)
+			encoding=`echo ${OPTARG} | sed s/\'/\'\'/g`
+			options="${options}, ENCODING '${encoding}'"
+			;;
+		d)
+			delimiter=`echo ${OPTARG} | sed s/\'/\'\'/g`
+			options="${options}, DELIMITER '${delimiter}'"
+			;;
+		e)
+			escape=`echo ${OPTARG} | sed s/\'/\'\'/g`
+			options="${options}, ESCAPE '${escape}'"
+			;;
+		H)
+			options="${options}, HEADER true"
+			;;
+		h)
+			usage 0 1 # normal status, STDOUT
+			;;
+		n)
+			null=`echo ${OPTARG} | sed s/\'/\'\'/g`
+			options="${options}, NULL '${null}'"
+			;;
+		q)
+			quote=`echo ${OPTARG} | sed s/\'/\'\'/g`
+			options="${options}, QUOTE '${quote}'"
+			;;
+		s)
+			schema=`echo ${OPTARG} | sed s/\'/\'\'/g`
+			;;
+		T)
+			format='text'
+			;;
+		*)
+			echo "$0: illegal option -- ${OPTARG}" >&2
+			usage 64 2 # EX_USAGE, STDERR
```

Comment:
Sure. I need to brush up on scoping rules, but not a bad idea.

BTW, there is no real "standard" for exit codes, except that BSD systems specify a few of them in `sysexit`. I believe the only real standard says "exit between 1 and 126 for error, 0 otherwise", so this is all kind of arbitrary.

At any rate, I like using the BSD ones for things that make sense (usage, missing files, etc.) and make up my own for the rest. At minimum, I consider something well-behaved if:

  * All of the output a user would care about goes to stdout
  * _No_ diagnostic or error output goes to stdout
  * It _always_ exits zero on success (I've seen tools that return counts as their status, which is insane)

Just trying to keep that standard for composability, etc.